.buildkite/pipeline.yml

```diff
@@ -3,16 +3,20 @@ steps:
     command: |
       nix-build -A lint -o lint.sh
       ./lint.sh
-    timeout: 100
-  - label: Build and test on GHC 8.4
-    command: |
-      nix-build -A monad-bayes-ghc84 --no-out-link
-    timeout: 100
-  - label: Build and test on GHC 8.6
-    command: |
-      nix-build -A monad-bayes-ghc86 --no-out-link
-    timeout: 100
+    timeout_in_minutes: 100
   - label: Build and test on GHC 8.8
     command: |
       nix-build -A monad-bayes-ghc88 --no-out-link
-    timeout: 100
+    timeout_in_minutes: 100
+  - label: Build and test on GHC 8.10
+    command: |
+      nix-build -A monad-bayes-ghc810 --no-out-link
+    timeout_in_minutes: 100
+  - label: Build and test on GHC 9
+    command: |
+      nix-build -A monad-bayes-ghc9 --no-out-link
+    timeout_in_minutes: 100
+  - label: Build a shell environment
```

Comment:
In [`mkMonadBayes` we only create a `shellFor`](https://github.com/tweag/monad-bayes/pull/122/files#diff-245563af9e34754739b12b294f3324c1d42c116ace7ce8735e21dfbc82d8e064R12) attribute which is a project function for constructing a shell environment specific to that ghc version. 

If we were to move the constructed [shell](https://github.com/tweag/monad-bayes/pull/122/files#diff-245563af9e34754739b12b294f3324c1d42c116ace7ce8735e21dfbc82d8e064R47) to be an attribute of the attribute set created by `mkMonadBayes`, then you would be absolutely correct.

Reply:
Then I don't understand why bind `"shellFor"` to a function in `recurseIntoAttrs`. What is it used for?

Reply:
It's there to create the `shell` attribute in `default.nix`, which is used by `shell.nix`. `mkMonadBayes` returns a subset of the attributes in the haskell.nix project. 

If the question is more about, "why is the `default.nix` structured this way?" then I don't have an answer to that. I mostly tried to leave it as I found it, just updating what was happening behind the scenes. The way I saw it, I had three choices: leave the `fetch*` for nixpkgs and haskell.nix as they were or switch to some managed solution. I chose to switch to a managed solution, which means I had to choose between niv and flakes. I chose flakes because they seem the most future proof.

Reply:
> It's there to create the shell attribute in default.nix, which is used by shell.nix. mkMonadBayes returns a subset of the attributes in the haskell.nix project.

But `defaultProject` is defined as

```nix
defaultProject = projects.monad-bayes-ghc810;
```

So whatever is in there comes from `./nix/projects.nix` and not from an application of `mkMonadBayes`. This is what confused my.

Reply:
Oooh I see now. Sorry, you are absolutely correct and `shellFor` in `mkMonadBayes` can be removed. I forgot I had refactored that to not be used any longer. I will remove it.

Reply:
Oooh I see now. Sorry, you are absolutely correct and shellFor in mkMonadBayes can be removed. I forgot I had refactored that to not be used any longer. I have removed it.